presto-main/src/main/java/io/prestosql/execution/SqlQueryManager.java

```diff
@@ -232,6 +236,8 @@ public void createQuery(QueryExecution queryExecution)
             finally {
                 // execution MUST be added to the expiration queue or there will be a leak
                 queryTracker.expireQuery(queryExecution.getQueryId());
+                // query MUST be removed from dynamic filter service or there will be a leak
+                dynamicFilterService.removeQuery(queryExecution.getQueryId());
```

Comment:
remove, use listener instead in `SqlQueryExecution`

Reply:
removed now